genesis/camino_genesis_test.go

```diff
@@ -0,0 +1,290 @@
+					thisConfig.Camino.Allocations = append(thisConfig.Camino.Allocations, KopernikusConfig.Camino.Allocations...)
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("consortium member not kyc verified"),
+		},
+		"wrong validator duration": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					thisConfig.Camino.Allocations = []CaminoAllocation{{AddressStates: AddressStates{ConsortiumMember: true, KYCVerified: true}, PlatformAllocations: []PlatformAllocation{{NodeID: ids.GenerateTestNodeID(), ValidatorDuration: 0}}}}
+					thisConfig.Camino.Allocations = append(thisConfig.Camino.Allocations, KopernikusConfig.Camino.Allocations...)
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("wrong validator duration"),
+		},
+		"repeated staker allocation": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					thisConfig.Camino.Allocations = []CaminoAllocation{{AddressStates: AddressStates{ConsortiumMember: true, KYCVerified: true}, PlatformAllocations: []PlatformAllocation{{NodeID: KopernikusConfig.Camino.Allocations[0].PlatformAllocations[0].NodeID, ValidatorDuration: 1}}}}
+					thisConfig.Camino.Allocations = append(thisConfig.Camino.Allocations, KopernikusConfig.Camino.Allocations...)
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("repeated staker allocation"),
+		},
+		"no staker allocations": {
+			args: args{
+				config: func() *Config {
+					thisConfig := KopernikusConfig
+					thisConfig.Camino.Allocations = []CaminoAllocation{{PlatformAllocations: []PlatformAllocation{}}}
+					return &thisConfig
+				}(),
+			},
+			err: errors.New("no staker allocations"),
+		},
+		"valid": {
+			args: args{
+				config: &KopernikusConfig,
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := validateCaminoConfig(tt.args.config)
+
+			if tt.err != nil {
+				require.ErrorContains(t, err, tt.err.Error())
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestBuildCaminoGenesis(t *testing.T) {
+	ctx := snow.DefaultContextTest()
+	rewards := reward.NewCalculator(defaultRewardConfig)
+	ctx.NetworkID = constants.KopernikusID
+	baseDB := memdb.New()
+	db := versiondb.New(baseDB)
+	cfg := defaultConfig()
+
+	addrs := set.Set[ids.ShortID]{}
+	_, _, avaxAddrBytes, _ := address.Parse(initialAdmin)
+	avaxAddr, _ := ids.ToShortID(avaxAddrBytes)
+	addrs.Add(avaxAddr)
+	outputOwners := secp256k1fx.OutputOwners{
+		Locktime:  0,
+		Threshold: 1,
+		Addrs:     []ids.ShortID{avaxAddr},
+	}
+	type args struct {
+		config *Config
+		hrp    string
+	}
+	tests := map[string]struct {
+		args          args
+		expectedUtxos map[ids.ID]*avax.UTXO
+		err           error
+	}{
+		"success - kopernikus": {
+			args: args{
+				config: &KopernikusConfig,
```

Comment:
Does kopernikusConfig contains all required cases to test? Like, all 4 type of p-chain allocations, 1 x-chain allocation
We need to test that `buildCaminoGenesis` correctly generates all new and old features, cause we also override avax x-chain generation
The most critical ones would be new/changed features, ofc, like validators txs, deposit txs, address states, msig aliases, deposit offers

Reply:
I can add further test cases with a separate PR after the holidays.